src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -163,4 +161,44 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
             s"from available source columns '${dfColumnNames.mkString(",")}'")
     }
   }
+
+  private def getIndexDataFrame(
+      spark: SparkSession,
+      df: DataFrame,
+      indexConfig: IndexConfig): (DataFrame, Seq[String], Seq[String]) = {
+    val (resolvedIndexedColumns, resolvedIncludedColumns) = resolveConfig(df, indexConfig)
+    val indexConfigColumns = resolvedIndexedColumns ++ resolvedIncludedColumns
+    val addLineage = spark.sessionState.conf
+      .getConfString(
+        IndexConstants.INDEX_LINEAGE_ENABLED,
+        IndexConstants.INDEX_LINEAGE_ENABLED_DEFAULT)
+      .toBoolean
+
+    val indexDF = if (addLineage) {
+      // Lineage is captured using two sets of columns:
+      // 1. DATA_FILE_NAME_COLUMN column contains source data filename for each index record.
+      // 2. If source data is partitioned, all partitioning key(s) are added to index schema.
```

Comment:
`// 2. If source data is partitioned, all partitioning key(s) are added to index schema as columns if they are not already part of the schema.`

Reply:
Done.